### DISCUSSION TITLE
Loading ESM-2 model by GPU

### DISCUSSION THREAD

**Discussion by ptynecki** [facebookresearch/esm#258] [Q&A]
Hello,

I would like to load **ESM-2** model by **GPU** to speed-up embedding execution. I have an access to RTX A6000 devices on my machine (48GB vRAM each instance).

I tried to use the same code base as it was recommended for ESM-1x models but the exception was raised:

```
CUDA out of memory. Tried to allocate 400.00 MiB (GPU 0; 47.54 GiB total capacity; 46.49 GiB already allocated; 321.06 MiB free; 46.50 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
```
As it was suggested, I tries to modify `PYTORCH_CUDA_ALLOC_CONF` value but it wasn't helpful.

Furthermore, I tied to adopt the [esm2_infer_fairscale_fsdp_cpu_offloading.py](https://github.com/facebookresearch/esm/blob/main/examples/esm2_infer_fairscale_fsdp_cpu_offloading.py) code but it seems >128 GB of RAM are required.

Happy to hear more tips for the issue.

Regards,
Piotr

**Comment by ptynecki** [comment]
I would be grateful if @tomsercu @nikitos9000 or other ESM core developers could make a comment for this issue.

**Comment by tomsercu** [comment]
Hi @ptynecki , actually 48GB RAM should be enough to load the largest 15B ESM-2 model in fp16 (would need 30GB for params, so you have 18GB left for internal state). 
Did you try to set the model to float16 before transfering to device?
